controllers/argocd/applicationset.go

```diff
@@ -58,9 +62,34 @@ func getArgoApplicationSetCommand(cr *argoproj.ArgoCD) []string {
 		cmd = append(cmd, ApplicationSetGitlabSCMTlsCertPath)
 	}
 
+	// appset source namespaces should be subset of apps source namespaces
+	appsetsSourceNamespaces := []string{}
+	appsNamespaces, err := r.getSourceNamespaces(cr)
+	if err == nil {
+		for _, ns := range cr.Spec.ApplicationSet.SourceNamespaces {
+			if contains(appsNamespaces, ns) {
+				appsetsSourceNamespaces = append(appsetsSourceNamespaces, ns)
+			}
```

Comment:
@svghadi we should probably have an else block here that logs an info/debug msg that a namespace was skipped because it has not been whitelisted as an app source ns

Reply:
There is already a message logged during resource creation in target namespaces. Added a debug log for this.